src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -617,17 +631,60 @@ internal override bool IsRecord
         }
 
 #nullable enable
-        internal sealed override bool IsExtension => false; // PROTOTYPE Extension symbols not yet supported in metadata
+        internal sealed override bool IsExtension => TypeKind == TypeKind.Extension;
+
+        internal sealed override TypeSymbol? ExtensionUnderlyingTypeNoUseSiteDiagnostics
+        {
+            get
+            {
+                return GetDeclaredExtensionUnderlyingType();
```

Comment:
>return GetDeclaredExtensionUnderlyingType(); [](http://example.com/codeflow?start=16&length=44)

Why do we need a property and a method which do the same thing, both `internal`?

Reply:
The reason for the separation is that for other properties (like bases or base extensions), we need to separate the property (which adds logic for cycles) from the method (which only tells you what is declared). So I was following the same model here, but it turns out we don't need to deal with cycles.